### PR TITLE
Add a customizable title in the webapp

### LINKED ISSUE #118: WebApp - Add title
**Description of Problem**:
Add a title at the top of Shapash WebApp

**Overview of the Solution**:
- add an attribute 'model_title' to the SmartExplainer Object
- model_title could be specified by Shapash User using compile() or add() method (add a new parameter)
- In the App code: manage truncation of a too long title (there is already a function in util.py to manage that kind of problem)

### PR DESCRIPTION
**Pull request by guillaume-vignal**
# Description

You van add now a title in the web app. You can test it with tutorial-01.

Fixes #118 

## Type of change

- [x] New feature

# How Has This Been Tested?

- [x] Test with no title in tutorial-01
- [x] Test with a title in tutorial-01

**Test Configuration**:
* OS: linux
* Python version: 3.8
* Shapash version: master

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

title_story
title_story_long

### CODE REVIEW

**Review by yg79 — CHANGES_REQUESTED** [review]
- Truncate the title when it is too long
- Allow user to add title using compile() or add()

**Review by yg79 — CHANGES_REQUESTED** [review]
- don't truncate attribute title_story
- truncate the title_story just before display it
- modify tutorial01-Shapash-Overview-Launch-WebApp.rst (title)
- modify webapp_launch_DVF.py (add title)

File: shapash/explainer/smart_plotter.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 from shapash.utils.utils import add_line_break, truncate_str, compute_digit_number, add_text, \
     maximum_difference_sort_value, compute_sorted_variables_interactions_list_indices
 
+
```
Comment (yg79) [inline shapash/explainer/smart_plotter.py:17]: too many /n

File: shapash/explainer/smart_plotter.py [hunk 2]
```diff
@@ -452,15 +461,17 @@ def plot_violin(self,
 
             else:
                 fig.add_trace(go.Violin(x=feature_values.loc[feature_values.iloc[:, 0] == i].values.flatten(),
-                                        y=contributions.loc[feature_values.iloc[:, 0] == i].values.flatten(),
+                                        y=contributions.loc[feature_values.iloc[:, 0]
+                                                            == i].values.flatten(),
```
Comment (yg79) [inline shapash/explainer/smart_plotter.py:465]: the code become difficult to read

File: shapash/explainer/smart_plotter.py [hunk 3]
```diff
@@ -452,15 +461,17 @@ def plot_violin(self,
...
                                         line_color=self.default_color,
                                         showlegend=False,
                                         meanline_visible=True,
                                         scalemode='count',
-                                        hovertext=hv_text_df.loc[feature_values.iloc[:, 0] == i].values.flatten(),
-                                        hovertemplate='<b>%{hovertext}</b><br />' + hv_temp,
-                                        customdata=contributions.index.values
-                                        ))
+                                        hovertext=hv_text_df.loc[feature_values.iloc[:, 0] == i].values.flatten(
```
Comment (yg79) [inline shapash/explainer/smart_plotter.py:470]: code become difficult to read with flatten(
),

File: shapash/explainer/smart_plotter.py [hunk 4]
```diff
@@ -1243,11 +1258,11 @@ def features_importance(self,
             total_len = self.explainer.x_pred.shape[0]
             addnote = add_text([addnote,
                                 f"Subset length: {subset_len} ({int(np.round(100 * subset_len / total_len))}%)"],
-                                sep=" - ")
+                               sep=" - ")
         if self.explainer.x_pred.shape[1] >= max_features:
             addnote = add_text([addnote,
                                 f"Total number of features: {int(self.explainer.x_pred.shape[1])}"],
-                                sep=" - ")
+                               sep=" - ")
```
Comment (yg79) [inline shapash/explainer/smart_plotter.py:1265]: there doesn't seem to be the same indentation as the row above

**Review by yg79 — CHANGES_REQUESTED** [review]
model.py have changed, this is not the last version, can you update your repo and your branch

File: shapash/explainer/smart_plotter.py (hunk 3 above)
Comment (guillaume-vignal) [reply shapash/explainer/smart_plotter.py:470]: It's not coming from myside

File: shapash/explainer/smart_plotter.py (hunk 4 above)
Comment (guillaume-vignal) [reply shapash/explainer/smart_plotter.py:1265]: True, I will change it

File: shapash/explainer/smart_plotter.py (hunk 1 above)
Comment (guillaume-vignal) [reply shapash/explainer/smart_plotter.py:17]: I didn't get it

File: shapash/explainer/smart_plotter.py (hunk 2 above)
Comment (guillaume-vignal) [reply shapash/explainer/smart_plotter.py:465]: The question is do we respect rules or not ? what is the guideline to follow ?

Comment (guillaume-vignal) [reply shapash/explainer/smart_plotter.py:465]: I tried to fix it

File: shapash/explainer/smart_plotter.py [hunk 5]
```diff
@@ -133,7 +134,8 @@ def __init__(self, explainer):
 
         self.round_digit = None
 
-        self.interactions_col_scale = ["rgb(175, 169, 157)", "rgb(255, 255, 255)", "rgb(255, 77, 7)"]
+        self.interactions_col_scale = [
+            "rgb(175, 169, 157)", "rgb(255, 255, 255)", "rgb(255, 77, 7)"]
```
Comment (yg79) [inline shapash/explainer/smart_plotter.py:138]: can you please modify this row

File: shapash/explainer/smart_plotter.py [hunk 6]
```diff
@@ -399,15 +404,18 @@ def plot_violin(self,
...
         hv_text_df = pd.DataFrame(hv_text, columns=['text'], index=feature_values.index)
         hv_temp = f'{feature_name} :<br />' + '%{x}<br />Contribution: %{y:.4f}<extra></extra>'
 
         # add break line to X label
-        max_len_by_row = max([round(50 / self.explainer.features_desc[feature_values.columns.values[0]]), 8])
-        feature_values.iloc[:, 0] = feature_values.iloc[:, 0].apply(add_line_break, args=(max_len_by_row, 120,))
+        max_len_by_row = max(
+            [round(50 / self.explainer.features_desc[feature_values.columns.values[0]]), 8])
+        feature_values.iloc[:, 0] = feature_values.iloc[:, 0].apply(
+            add_line_break, args=(max_len_by_row, 120,))
```
Comment (yg79) [inline shapash/explainer/smart_plotter.py:418]: generate a loss of clarity

File: shapash/explainer/smart_plotter.py (hunk 2 above)
Comment (yg79) [reply shapash/explainer/smart_plotter.py:464]: generate a loss of clarity

File: shapash/explainer/smart_plotter.py (hunk 3 above)
Comment (yg79) [reply shapash/explainer/smart_plotter.py:469]: generate a loss of clarity

File: shapash/explainer/smart_plotter.py (hunk 4 above)
Comment (yg79) [reply shapash/explainer/smart_plotter.py:1263]: ok

File: shapash/explainer/smart_plotter.py [hunk 7]
```diff
@@ -1820,7 +1839,8 @@ def interactions_plot(self,
             feature_values1 = self.explainer.x_pred.loc[list_ind, col_name1].to_frame()
             feature_values2 = self.explainer.x_pred.loc[list_ind, col_name2].to_frame()
 
-        interaction_values = self.explainer.get_interaction_values(selection=list_ind)[:, col_id1, col_id2]
+        interaction_values = self.explainer.get_interaction_values(selection=list_ind)[
+            :, col_id1, col_id2]
```
Comment (yg79) [inline shapash/explainer/smart_plotter.py:1843]: generate a loss of clarity

File: shapash/explainer/smart_plotter.py [hunk 8]
```diff
@@ -1905,11 +1925,13 @@ def top_interactions_plot(self,
...
 
-        list_ind, addnote = self._select_indices_interactions_plot(selection=selection, max_points=max_points)
+        list_ind, addnote = self._select_indices_interactions_plot(
+            selection=selection, max_points=max_points)
 
         interaction_values = self.explainer.get_interaction_values(selection=list_ind)
 
-        sorted_top_features_indices = compute_sorted_variables_interactions_list_indices(interaction_values)
+        sorted_top_features_indices = compute_sorted_variables_interactions_list_indices(
+            interaction_values)
```
Comment (yg79) [inline shapash/explainer/smart_plotter.py:1934]: generate a loss of clarity

**Review by yg79 — CHANGES_REQUESTED** [review]
some modifications to improve readability

**Review by yg79 — APPROVED** [review]

**Pull request merged**